### PR TITLE
s3 changes for datadelivery-cli

### PR DESCRIPTION
**Pull request by johnbradley**
Adds name field to endpoint model.
Adds and ability filter `/s3-endpoints/` by name.
Adds the ability to filter `/s3-buckets/` by name.
Adds the ability to filter `/s3-users/` by endpoint and user.

### CODE REVIEW

**Comment by johnbradley** [comment]
The current S3User only exists for a specific endpoint. Right now this is a sub-route of endpoints: `/s3-endpoints/<id>/current-user/`. This seems weird for some reason. I am considering create a `/users/current-user/` endpoint then allowing a user to filter `/s3-users` by endpoint.id and user.id.

**Comment by johnbradley** [comment]
With the current implementation the user serializer always fetches the users DukeDS id:
https://github.com/Duke-GCB/D4S2/blob/d344156c5ed08e080e744d367da250056132085e/d4s2_api_v2/serializers.py#L47-L57

With out creating a OAuthService in the admin this fails with a 500 error: 
```
AttributeError at /api/v2/users/current-user/
'NoneType' object has no attribute 'client_id'
```

**Comment by dleehr** [comment]
> With out creating a OAuthService in the admin this fails with a 500 error:

As I'm working on the changes for s3-ownership, I'm also adding some helpers to gcb-web-auth to better handle this: https://github.com/Duke-GCB/gcb-web-auth/commit/20fefe73ac0ee5d21aeec89bfa58d22895b5203a

This would raise an `OAuthUnconfiguredException`, which you could catch.

I don't know how much that helps here, but it's also worth asking if we need the dds id in this serializer..

**Comment by johnbradley** [comment]
> it's also worth asking if we need the dds id in this serializer

I like the idea of removing this field (`duke_ds_user`) from the users endpoint. Even if I can fetch the DukeDS ID without an error it seems like a waste of time to read data from the DukeDS API when I really just need the django current user id. 

We are using this field in datadelivery-ui here:
https://github.com/Duke-GCB/datadelivery-ui/blob/9c9fab537a18f4c278a008829392944c925453d4/app/models/user.js#L8

**Comment by dleehr** [comment]
Let's remove the field for now, and keep this endpoint/serializer focused on D4S2 data.

File: d4s2_api_v2/serializers.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from rest_framework import serializers
 from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
```
Comment (dleehr) [inline d4s2_api_v2/serializers.py]: I don't see this used

**Pull request merged**